annotation/tests/test_annotations.py

```diff
@@ -83,3 +172,142 @@ def test_row_to_dict(row, expected_dictionary):
     result = row_to_dict(row)
     print(result)
     assert result == expected_dictionary
+
+
+@pytest.mark.unittest
+@pytest.mark.parametrize(
+    [
+        "s3_prefix",
+        "bucket_name",
+        "expected_string",
+    ],
+    [
+        (
+            "S3_test",
+            "bucket_test",
+            "S3_test-bucket_test",
+        ),
+        (
+            None,
+            "bucket_test",
+            "bucket_test",
+        ),
+    ],
+)
+def test_convert_bucket_name_if_s3prefix(
+    monkeypatch,
+    s3_prefix,
+    bucket_name,
+    expected_string,
+):
+    monkeypatch.setattr("annotation.annotations.main.S3_PREFIX", s3_prefix)
+    result = convert_bucket_name_if_s3prefix(bucket_name=bucket_name)
+    assert result == expected_string
+
+
+@pytest.mark.unittest
+@pytest.mark.parametrize(
+    [
+        "s3_provider",
+        "bucket_name",
+    ],
+    [
+        (
+            "minio",
+            TEST_TENANT,
+        ),  # successful connection
+        (
+            "aws_iam",
+            TEST_TENANT,
+        ),  # successful connection
+        (
+            "no_provider",
+            TEST_TENANT,
+        ),  # NotConfiguredException
+        (
+            "aws_iam",
+            "no_bucket",
+        ),  # NoSuchBucket
```

Comment:
Let's separate bad cases into another test and cover them there. This way your parametrize for good cases wont need TEST_TENANT (it doesn't change)

Reply:
Done